frontend/packages/client/src/components/ProposalsList/ProposalCardFooter.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useMemo, useState } from "react";
+import millify from "millify";
+import { Active, CheckCircle } from "../Svg";
+import { parseDateFromServer } from "utils";
+import StatusLabel from "../StatusLabel";
+import { useVotingResults } from "hooks";
+import { FilterValues } from "const";
+
+const ProposalCardFooter = ({ id, voted, endTime, computedStatus, isDesktopOnly }) => {
+  const [justMounted, setJustMounted] = useState(true);
+
+  const { getVotingResults, data: votingResults } = useVotingResults();
+
+  const { diffDuration } = parseDateFromServer(endTime);
+
+  const status = FilterValues[computedStatus] ?? FilterValues.closed;
+
+  useEffect(() => {
+    if (justMounted && (FilterValues[computedStatus] === FilterValues.closed)) {
+      async function _getVotingResults() {
+        return getVotingResults(id);
+      }
+      _getVotingResults();
+    }
+  }, [computedStatus, getVotingResults, id, justMounted]);
+
+  // prevents memory leak in above useEffect for getVotingResults
+  useEffect(() => () => { setJustMounted(false) }, []);
+
+  const { textDecision, winCount } = useMemo(() => {
+    if (votingResults?.results) {
+      const { results } = votingResults;
+      const resultsArray = Object.entries(results);
+      const sortedResults = resultsArray.sort((a, b) => {
+        if (Number(b[1]) < Number(a[1])) {
+          return -1;
+        }
+        return 1;
+      });
+
+      const textDecision = sortedResults[0][0];
+      const winCount = Number(sortedResults[0][1]);
+
+      // no winner
+      if (sortedResults.every((e) => Number(e[1]) === winCount)) {
+        return { textDecision: "", winCount: 0 };
+      }
+
+      return {
+        textDecision,
+        winCount,
+      };
+    }
+    return { textDecision: "", winCount: 0 };
+  }, [votingResults]);
+
+  const iconStatusMap = {
+    [FilterValues.active]: <Active />,
+    [FilterValues.closed]:
+      textDecision !== "" ? <CheckCircle width="15" height="15" /> : null,
+  };
+
+  const textDescriptionMap = {
+    [FilterValues.active]: `${diffDuration} left`,
+    [FilterValues.closed]:
+      textDecision !== ""
+        ? `${textDecision} ${winCount !== 0 ? `(${millify(winCount)})` : ""}`
+        : "",
+  };
+
+  const statusLabelMap = {
+    [FilterValues.active]: <StatusLabel status="Active" voted={voted} rounder
+      color="has-background-orange"
+      className="proposal-status-label has-text-weight-bold has-text-black"
+    />,
+    [FilterValues.pending]: <StatusLabel status="Pending" voted={voted} rounder
+      color="has-background-grey-light"
+    />,
+    [FilterValues.closed]: <StatusLabel status="Cancelled" voted={voted} rounder />,
+    [FilterValues.cancelled]: <StatusLabel status="Cancelled" voted={voted} rounder />,
+  };
+
+  // gross section to handle styling
+  let extraClasses1 = "";
+  let extraClasses2 = "";
+  let extraStyles1 = {};
+  let extraClasses3 = "";
+  let extraClasses4 = "";
+
```

Comment:
there's a WrapperComponent that handles styles from mobile and desktop https://github.com/DapperCollectives/CAST/blob/c37ce0380c3b9e71a72f7c753aed1b8f29b3d0ef/frontend/packages/client/src/components/WrapperResponsive.js 
maybe can be used to simplify the if clause

Reply:
Anytime you find yourself in a situation where you use generic variable names like these it's a good signal to think a little more about your naming convention. Naming things is one of the hardest parts of programming. 

In this situation you could use names like `Container`, `Icon`, `Status`, `Content`, etc. Try to use a name that describes a little about the element where the class or style is being applied.